python_modules/dagster/dagster/_core/definitions/declarative_automation/automation_condition.py

```diff
@@ -448,7 +450,12 @@ def __init__(
         self._subsets_with_metadata = check.opt_sequence_param(
             subsets_with_metadata, "subsets_with_metadata", AssetSubsetWithMetadata
         )
-        self._extra_state = extra_state
+        check.param_invariant(
+            not (cursor and extra_state),
+            "extra_state",
+            "Cannot provide both cursor and extra_state.",
+        )
+        self._extra_state = check.opt_str_param(cursor, "cursor") or extra_state
```

Comment:
<!--__GRAPHITE_HTML_TAG_START__--><i class='graphite__hidden'>[Re: lines +432 to +458]<br/><br/></i>

<!--__GRAPHITE_HTML_TAG_END__-->I think it's pretty non-obvious as to why these should be mutually exclusive given the names etc<!--__GRAPHITE_HTML_TAG_START__--><span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23596?utm_source=unchanged-line-comment">Graphite</a>.</span>

<!--__GRAPHITE_HTML_TAG_END__-->